content/cucumber/api.md

```diff
@@ -343,9 +343,32 @@ end
 {{% block "javascript" %}}Cucumber.js does not support `Around` hooks.{{% /block %}}
 
 ## Step hooks
-
+{{% text "java" %}}
+Step hooks invoked before and after a step. The hooks have 'invoke around' semantics. Meaning that if a `BeforeStep`
+hook is executed the `AfterStep` hooks will also be executed regardless of the result of the step. If a step did not 
+pass, the following step and its hooks will be skipped.
+{{% /text %}}
 ### BeforeStep
 
+{{% text "ruby" %}}Cucumber-Ruby does not support `BeforeStep` hooks.{{% /text %}}
```

Comment:
Drive-by question: Is this section in need of `{{%block "java"}}` and so on? (I am only comparing to AfterStep section.)

Reply:
Ah. Yes. Forgot it.